### PR TITLE
Unit Tests and Rework Testing Framework

### PR DESCRIPTION
**Pull request by Dtphelan1**
Moves old tests to an integration file, and creates new unit tests. Establishes a directory structure for testing that separates those types of tests.  Currently there are two integration tests failling, so MG and I will reach out to DM for clarification if needed.

### CODE REVIEW

**Comment by kjmahalingam** [comment]
The integration tests were ancient. I've removed them since they were very outdated and depended on an old infrastructure locally that we no longer have. The unit tests should cover what we need, and we have the client to use for end-to-end testing if we desire that.

I did a pass of the tests, but will look again now in more detail so that I can review. First glance is that this all looks good. :)

**Comment by kjmahalingam** [comment]
The unit tests all look great to me!
I noticed that the pieces fitting together seem to behave a little differently in the lambdas than they originally did. I tried deploying them and I think that something went a little awry with the proxy lambda and the messaging lambda since they are preventing authorization or preventing message processing, respectively.
Once we are able to fix the actual *lambda* implementation itself to be back in order as before (I think the modularizing messed something up), the unit tests do test the individual components as I would expect!

File: proxy/index.js [hunk 1]
```diff
@@ -1,31 +1,46 @@
...
+  return process.env.OAUTH_SERVER_PATH + '?' + querystring.stringify(params);
 };
 
-exports.handler = async (event) => {
+/**
+ * Generate request options based on env variables and search params
+ * @param {string[]} params - the request query params
+ * @return {Object} An options object for requests
+ */
+function generateOptions(params) {
```
Comment (kjmahalingam) [inline proxy/index.js:23]: One more thought; could we write one quick (pretty small) test for this function, to verify that the output of it given an input is what we expect?

**Comment by kjmahalingam** [comment]
Fixed the proxy lambda; there was a node module added to it, but the build script did not yet include node modules for that lambda. That has been pushed up now.

Working on the messaging lambda now.

File: process-message/index.js [hunk 2]
```diff
@@ -17,55 +45,47 @@ exports.handler = async (event, context, callback) => {
...
     callback(responses.response400(
         'Message does not contain Bundle.',
     ));
     return;
   }
   console.log('Collected MessageHeader and Bundle.');
 
-  const researchSubject =
-    getBundleResourcesByType(bundle, 'ResearchSubject', {}, true);
+  const researchSubject = getBundleResourcesByType(bundle, 'ResearchSubject', {}, true);
```
[the remark this answers is not in the document]
Comment (Dtphelan1) [reply process-message/index.js]: This should be resolved!

File: process-message/index.js [hunk 3]
```diff
@@ -17,55 +45,47 @@ exports.handler = async (event, context, callback) => {
...
   if (!researchSubject) {
     callback(responses.response400(
         'Bundle does not contain ResearchSubject.',
     ));
     return;
   }
-  const researchStudy =
-    getBundleResourcesByType(bundle, 'ResearchStudy', {}, true);
+  const researchStudy = getBundleResourcesByType(bundle, 'ResearchStudy', {}, true);
```
[the remark this answers is not in the document]
Comment (Dtphelan1) [reply process-message/index.js]: This should be resolved!

**Pull request merged**